bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/ControllerHandler.java

```diff
@@ -146,25 +151,19 @@ public boolean isConnected() {
                 logger.trace("Sending command '{}'", command);
                 OutputStream out = localSocket.getOutputStream();
                 out.write(command.getBytes());
-                out.write("\r\n".getBytes());
+                out.write(LF);
 
-                try (Reader isr = new InputStreamReader(localSocket.getInputStream());
-                        BufferedReader in = new BufferedReader(isr)) {
-                    while (true) {
-                        String line = in.readLine();
-                        logger.trace("Read result '{}'", line);
-                        if ("OK".equals(line)) {
-                            return response.toString();
-                        }
-                        response.append(line);
-                        if (response.length() > 100) {
-                            /*
-                             * Usually this loop times out on errors, but in the case that we just keep getting
-                             * data we should also fail with an error.
-                             */
-                            throw new CoolMasterClientError(
-                                    String.format("Got gibberish response to command %s", command));
-                        }
+                final Reader isr = new InputStreamReader(localSocket.getInputStream());
+                final BufferedReader in = new BufferedReader(isr);
+                while (true) {
+                    String line = in.readLine();
```

Comment:
`in.readLine()` will return null if the end of stream is reached. If that happens your code here will go into an infinite loop.

Reply:
The original code didn't handle null, so I didn't either. But I've fixed it and will push a commit.